core-bundle/src/Search/Indexer/DefaultIndexer.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Search\Indexer;
+
+use Contao\Automator;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\Search\Document;
+use Contao\Search;
+
+class DefaultIndexer implements IndexerInterface
+{
+    /**
+     * @var ContaoFramework
+     */
+    private $framework;
+
+    /**
+     * @var bool
+     */
+    private $indexProtected = false;
+
+    public function __construct(ContaoFramework $framework, bool $indexProtected = false)
+    {
+        $this->framework = $framework;
+        $this->indexProtected = $indexProtected;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function index(Document $document): void
+    {
+        if (200 !== $document->getStatusCode()) {
+            return;
+        }
+
+        $meta = [
+            'title' => 'undefined',
+            'language' => 'en',
+            'protected' => false,
+            'groups' => [],
+            'pageId' => 0,
+            'noSearch' => true, // Causes the indexer to skip this document if there is no json-ld data
+        ];
+
+        $this->extendMetaFromJsonLdScripts($document, $meta);
+
+        // If search was disabled in the page settings, we do not index
+        if (isset($meta['noSearch']) && true === $meta['noSearch']) {
+            return;
+        }
+
+        // If the front end preview is activated, we do not index
+        if (isset($meta['fePreview']) && true === $meta['fePreview']) {
+            return;
+        }
+
+        // If the page is protected, we only index if the member is logged in and protecting indexed pages is enabled
+        if (isset($meta['protected']) && true === $meta['protected']) {
+            if (!$this->indexProtected) {
+                return;
+            }
+        }
+
+        $this->framework->initialize();
+
+        /** @var Search $search */
+        $search = $this->framework->getAdapter(Search::class);
+        $search->indexPage([
+                'url' => (string) $document->getUri(),
+                'content' => $document->getBody(),
+                'protected' => ($meta['protected']) ? '1' : '',
+                'groups' => $meta['groups'],
+                'pid' => $meta['pageId'],
+                'title' => $meta['title'],
+                'language' => $meta['language'],
+            ])
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear(): void
+    {
+        $this->framework->initialize();
+
+        /** @var Automator $automator */
+        $automator = $this->framework->getAdapter(Automator::class);
+        $automator->purgeSearchTables();
```

Comment:
Should we use `$indexer->clear()` in the `Automator::purgeSearchTables()` method instead and copy the code for truncating `tl_search` and `tl_search_index` to the `DefaultIndexer` here?

This would have the benefit that all enabled indexers get cleared if you clear the search index in the backend.

Reply:
Right, done in 1b715d395b939d6f9924d1f7a66e97f85c0de884.